### PR TITLE
Update Jenkins to Latest LTS fixes #731

### LINKED ISSUE #731: Jenkins Chart not using latest LTS
Hello,

The jenkins char doesn't use the latest lts which at the time of writing this ticket was 2.32.2.

### PR DESCRIPTION
**Pull request by jwhitcraft**
Also updated the plugins.

### CODE REVIEW

**Comment by jwhitcraft** [comment]
I signed it!

**Comment by lachie83** [comment]
@viglesiasce can you confirm if the containers will be automatically built?

**Comment by lachie83** [comment]
@k8s-bot ok to test

**Comment by scottrigby** [comment]
@jwhitcraft Just checking, it looks like `v0.7.0` isn't in the [registry](https://console.cloud.google.com/gcr/images/kubernetes-charts-ci/global/jenkins-master-k8s?project=kubernetes-charts-ci)?

> Failed to pull image "gcr.io/kubernetes-charts-ci/jenkins-master-k8s:v0.7.0": image pull failed for gcr.io/kubernetes-charts-ci/jenkins-master-k8s:v0.7.0, this may be because there are no credentials on this request. details: (Tag v0.7.0 not found in repository gcr.io/kubernetes-charts-ci/jenkins-master-k8s)
> Error syncing pod, skipping: failed to "StartContainer" for "copy-default-config" with ErrImagePull: "image pull failed for gcr.io/kubernetes-charts-ci/jenkins-master-k8s:v0.7.0, this may be because there are no credentials on this request. details: (Tag v0.7.0 not found in repository gcr.io/kubernetes-charts-ci/jenkins-master-k8s)"

**Comment by jwhitcraft** [comment]
@scottrigby I don't believe I have access to push that into the repo or even if i should be the one to do it.

Do you know where i can find documentation around it?

File: stable/jenkins/values.yaml
Comment (seanknox) [inline stable/jenkins/values.yaml:9]: I believe v0.6.0 is the latest version of this image (there's no v0.7.0).

**Review by seanknox — CHANGES_REQUESTED** [review]

**Comment by jwhitcraft** [comment]
@seanknox and @lachie83 

I just followed what was done in #341 where he updated the Dockerfile and also updated the image.  I'm not sure if @dserodio has access to push the image as well, but if given access i would be more than happy to build the image locally and push it to the registry.

**Edit** It looks like @viglesiasce did the image for that PR.

**Comment by lachie83** [comment]
@viglesiasce can you let us know the procedure to get a new image pushed to the repo?

**Comment by scottrigby** [comment]
Yep, I built a [test image](https://hub.docker.com/r/r6by/jenkins/) from the master Dockerfile, and it installed the latest LTS.

Of course we need to wait for the official image to merge this PR, but to see that it works you can override the image values:
```sh
helm install stable/jenkins \
--set Master.Image=r6by/jenkins,Master.ImageTag="0.1.1"
```

_Edit: simplified comment again._

File: stable/jenkins/master-image/Dockerfile
Comment (scottrigby) [inline stable/jenkins/master-image/Dockerfile]: Also update to latest jenkins git plugin `git:3.1.0`.

**Review by scottrigby — CHANGES_REQUESTED** [review]

**Comment by dserodio** [comment]
No @jwhitcraft, my PR refered to a non-existing image and @viglesiasce build and pushed that image afterwards.

File: stable/jenkins/values.yaml
Comment (seanknox) [reply stable/jenkins/values.yaml:9]: @jwhitcraft That image isn't publicly available. Would you be opening a separate PR/issue for that? That'd allow us to merge in the rest of the changes now.

```
$ docker pull gcr.io/kubernetes-charts-ci/jenkins-master-k8s:v0.7.0
Pulling repository gcr.io/kubernetes-charts-ci/jenkins-master-k8s
Tag v0.7.0 not found in repository gcr.io/kubernetes-charts-ci/jenkins-master-k8s
```

Comment (jwhitcraft) [reply stable/jenkins/values.yaml:9]: @seanknox, the rest of the changes don't make sense without having the updated docker image for the chart to pull down, IMO.

**Comment by jwhitcraft** [comment]
@dserodio that is what mine does, the 0.7.0 tag currently doesn't exist and i'm just waiting to someone with access to create and push the image up, so that PR can be merged.

**Comment by seanknox** [comment]
hey @viglesiasce, could you push up `gcr.io/kubernetes-charts-ci/jenkins-master-k8s:v0.7.0`?

File: stable/jenkins/values.yaml
Comment (scottrigby) [reply stable/jenkins/values.yaml:9]: @seanknox Agreed, they are valueless without updating the image. I have spoken with @lachie83 on Slack about the build process, and while I realize automating that with dependable security may be a longer-term effort, the fact that it has taken so long to do a simple image update means perhaps there should be more than one person with ability to do this. I would be happy to do so (as have done in my test image mentioned in [this comment](https://github.com/kubernetes/charts/pull/733#issuecomment-287483737). Is there someone else already on the team who can do this in the meantime, until the build process is reworked?

File: stable/jenkins/master-image/Dockerfile
Comment (scottrigby) [reply stable/jenkins/master-image/Dockerfile]: 👍

**Comment by dtzar** [comment]
If it's open - I'd be game to help move the gcr.io repo for this to an automated public Docker hub build linked to a GitHub repo backed by a community of people who want to support the image/chart to make image updates go quicker/easier...

**Comment by scottrigby** [comment]
@dtzar I spoke with @lachie83 on Slack, offering a similar thing (it's quite a long conversation, threaded in between others. For anyone interested, [see archives here](https://kubernetes.slack.com/archives/C0NH30761/p1490030121198993)). There is a very easy way to do this in gcr.io (set up a build trigger), but that's not desired by the maintainers.

Tl;dr, The current state is that the CI is being reworked/relocated, so that there is a proper security review for image creation. The last note I made about this. In the meantime people can build the image themselves, or use the image I created and steps mentioned in [my comment above](https://github.com/kubernetes/charts/pull/733#issuecomment-287483737).

**Comment by scottrigby** [comment]
Also, I agree the image could be built/pushed manually again in the meantime. Given how fast/easy that is to do, it would be a good thing to do for chart adoption even before the CI is reworked.

**Comment by scottrigby** [comment]
@jwhitcraft Can you also rebase this PR? `Chart.yaml` had an update in the meantime.

**Comment by viglesiasce** [comment]
I will push this up today.

**Comment by viglesiasce** [comment]
Image has been pushed. Sorry for the delay.

**Comment by jwhitcraft** [comment]
We might want to wait to merge this since a new lts was cut today for Jenkins.

**Comment by dtzar** [comment]
@jwhitcraft you'd have to wait for another image to be pushed, so IMO I'd just make sure the Jenkins Charts e2e passes and ship it.  You could do another PR for the new Jenkins lts :)

**Comment by jwhitcraft** [comment]
@k8s-bot e2e test this

**Review by seanknox — APPROVED** [review]

**Comment by seanknox** [comment]
LGTM. Let get this combined labor of love in! @lachie83  @mgoodness

**Comment by mgoodness** [comment]
Nice work, everyone! Merging.

**Pull request merged**
